lightning/src/ln/channel.rs

```diff
@@ -2448,16 +2474,10 @@ impl<Signer: Sign> Channel<Signer> {
 
 		// Update state now that we've passed all the can-fail calls...
 		let mut need_commitment = false;
-		if !self.is_outbound() {
-			if let Some(fee_update) = self.pending_update_fee {
-				self.feerate_per_kw = fee_update;
-				// We later use the presence of pending_update_fee to indicate we should generate a
-				// commitment_signed upon receipt of revoke_and_ack, so we can only set it to None
-				// if we're not awaiting a revoke (ie will send a commitment_signed now).
-				if (self.channel_state & ChannelState::AwaitingRemoteRevoke as u32) == 0 {
-					need_commitment = true;
-					self.pending_update_fee = None;
-				}
+		if let &mut Some((_, ref mut update_state)) = &mut self.pending_update_fee {
+			if *update_state == FeeUpdateState::RemoteAnnounced {
```

Comment:
Sure, could, also dunno if its worth a followup to add a debug assertion :)

Reply:
Narrator voice : it was.

Reply:
Ah, this is why I didn't:

```
error[E0502]: cannot borrow `*self` as immutable because it is also borrowed as mutable
    --> lightning/src/ln/channel.rs:2480:20
     |
2478 |         if let &mut Some((_, ref mut update_state)) = &mut self.pending_update_fee {
     |                                                       ---------------------------- mutable borrow occurs here
2479 |             if *update_state == FeeUpdateState::RemoteAnnounced {
2480 |                 debug_assert!(!self.is_outbound());
     |                                ^^^^ immutable borrow occurs here
2481 |                 *update_state = FeeUpdateState::AwaitingRemoteRevokeToAnnounce;
     |                 -------------------------------------------------------------- mutable borrow later used here
```